dali/python/nvidia/dali/plugin/mxnet.py

```diff
@@ -423,3 +459,224 @@ def __init__(self,
+                 Setting this flag to False will cause the iterator to return
+                 exactly 'size' entries.
+    last_batch_padded : bool, optional, default = False
+                 Whether the last batch provided by DALI is padded with the last sample
+                 or it just wraps up. In the conjunction with `fill_last_batch` it tells
+                 if the iterator returning last batch with data only partially filled with
+                 data from the current epoch is dropping padding samples or samples from
+                 the next epoch. If set to False next epoch will end sooner as data from
+                 it was consumed but dropped. If set to True next epoch would be the
+                 same length as the first one. For this happen, the option `pad_last_batch`
+                 in the reader need to be set to `True` as well.
+
+    Example
+    -------
+    With the data set ``[1,2,3,4,5,6,7]`` and the batch size 2:
+
+    fill_last_batch = False, last_batch_padded = True  -> last batch = ``[7]``, next iteration will return ``[1, 2]``
+
+    fill_last_batch = False, last_batch_padded = False -> last batch = ``[7]``, next iteration will return ``[2, 3]``
+
+    fill_last_batch = True, last_batch_padded = True   -> last batch = ``[7, 7]``, next iteration will return ``[1, 2]``
+
+    fill_last_batch = True, last_batch_padded = False  -> last batch = ``[7, 1]``, next iteration will return ``[2, 3]``
+
+    """
+    def __init__(self,
+                 pipelines,
+                 size,
+                 output_types=None,
+                 auto_reset=False,
+                 fill_last_batch=True,
+                 last_batch_padded=False):
+        super(DALIGluonIterator, self).__init__(
+            pipelines,
+            size,
+            fill_last_batch,
+            last_batch_padded,
+            auto_reset)
+
+        # Build all pipelines
+        for p in self._pipes:
+            with p._check_api_type_scope(types.PipelineAPIType.ITERATOR):
+                p.build()
+        self._data_batches = [None for i in range(self._num_gpus)]
+        self._counter = 0
+        self._output_tags = {DALIGluonIterator.DENSE_TAG, DALIGluonIterator.SPARSE_TAG}
+        assert output_types is None or set(output_types) <= self._output_tags, \
+            "Only DENSE_TAG and SPARSE_TAG are allowed"
+
+        self._outputs_types = output_types
+
+        # We need data about the batches (like shape information),
+        # so we need to run a single batch as part of setup to get that info
+        for p in self._pipes:
+            with p._check_api_type_scope(types.PipelineAPIType.ITERATOR):
+                p.schedule_run()
+        self._first_batch = None
+        self._first_batch = self.next()
+
+
+    def __next__(self):
+        if self._first_batch is not None:
+            batch = self._first_batch
+            self._first_batch = None
+            return batch
```

Comment:
The first_batch trick doesn't seem useful in this variant of Iterator. Can you remove it?
It seems it was carried over from DALIGenericIterator.